solr/modules/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java

```diff
@@ -234,6 +236,13 @@ protected static boolean scoreSingleHit(
 
     scorer.getDocInfo().setOriginalDocScore(hit.score);
     hit.score = scorer.score();
+    if (QueryLimits.getCurrentLimits()
+        .maybeExitWithPartialResults(
+            "Learning To Rank rescoring -"
+                + " The full reranking didn't complete."
+                + " If partial results are tolerated the reranking got reverted and all documents preserved their original score and ranking.")) {
```

Comment:
subjective, code complexity/simplicity vs. user/operational friendliness: the _"If partial results are tolerated"_ probably is 'knowable' at this point in the code i.e. if the info/knowledge were to passed down to this API then the wording could be differentiated here i.e. no 'if' qualifier in the wording.